c2h/catch2_runner_helper.inl

```diff
@@ -37,9 +37,10 @@
 int device_guard(int device_id)
 {
   int device_count{};
-  if (cudaGetDeviceCount(&device_count) != cudaSuccess)
+  auto err = cudaGetDeviceCount(&device_count);
+  if (err != cudaSuccess)
   {
-    std::cerr << "Can't query devices number." << std::endl;
+    std::cerr << "Error in cudaGetDeviceCount: " << cudaGetErrorString(err) << std::endl;
     std::exit(-1);
   }
```

Comment:
I believe this should rather be:
```suggestion
_CCCL_TRY_CUDA_API(::cudaGetDeviceCount, &device_count)
```

Reply:
That macro doesn't translate the error code into a human readable string, which was the motivation behind this change. I'd prefer to leave this as-is.

Reply:
Updated the API macros here: https://github.com/NVIDIA/cccl/pull/4588

Reply:
Updated to use new API macros.